bolt/lib/Rewrite/LinuxKernelRewriter.cpp

```diff
@@ -1126,6 +1134,27 @@ Error LinuxKernelRewriter::readParaInstructions() {
   return Error::success();
 }
 
+Error LinuxKernelRewriter::rewriteParaInstructions() {
+  // Disable output of functions with paravirtual instructions before the
+  // rewrite support is complete.
+  for (BinaryFunction &BF : llvm::make_second_range(BC.getBinaryFunctions())) {
+    if (!BC.shouldEmit(BF))
+      continue;
+    for (const BinaryBasicBlock &BB : BF) {
+      if (!BC.shouldEmit(BF))
+        break;
```

Comment:
```suggestion
```

Reply:
Why? That's intentional early break after we mark the function as non-simple in the loop below.

Reply:
Oh, that's subtle. Then I'd suggest breaking out immediately after `BF.setSimple(false)` to be explicit.

Reply:
It's breaking, just not to the outer loop. I've refactored using `llvm::any_of()` as you suggested and it looks much cleaner now.